04-GO/Dockerfile

```diff
@@ -1,22 +1,33 @@
-FROM golang:1.18 as build 
+# builder stage
+# from golang alpine image
+FROM golang:1.19-alpine as builder
 
+# Installing Git for alpine
+RUN apk add --no-cache git
+
+# create work directory called src
 WORKDIR /src
 
-COPY go.mod .
-# COPY go.sum .
+# copy go.mod and go.sum
+COPY go.mod go.sum ./
 
+# download dependencies
 RUN go mod download
 
+# copy other files
 COPY . .
 
-RUN CGO_ENABLED=0 GO111MODULE=on go build -v -o sample -installsuffix cgo -ldflags="-w -s" ./cmd/*.go
+# build golang app
+RUN CGO_ENABLED=0 GO111MODULE=on go build -v -o /main -installsuffix cgo -ldflags="-w -s" ./cmd/*.go
 
+# second stage
 FROM alpine
 
-COPY --from=build /src/sample /
+# copy executable file
+COPY --from=builder /main .
```

Comment:
why r u using `/` as prefix of your output?

Reply:
@jigsaw373 jan, because I was using ```/main``` when I was building the executable file.
If will fix it in the next commits.